landoapi/hg.py

```diff
@@ -368,56 +361,75 @@ def apply_patch(self, patch_io_buf):
                 + ["--logfile", f_msg.name]
             )
 
-    def format(self) -> Optional[List[str]]:
-        """Run `hg fix` to format the currently checked-out stack, reading
-        fileset patterns for each formatter from the `.lando.ini` file in-tree."""
-        # Avoid attempting to autoformat without `.lando.ini` in-tree.
-        lando_config_path = Path(self.path) / ".lando.ini"
-        if not lando_config_path.exists():
+    def read_lando_config(self) -> Optional[configparser.ConfigParser]:
+        """Attempt to read the `.lando.ini` file."""
+        try:
+            lando_ini_contents = self.read_checkout_file(".lando.ini")
+        except ValueError:
             return None
 
         # ConfigParser will use `:` as a delimeter unless told otherwise.
         # We set our keys as `formatter:pattern` so specify `=` as the delimiters.
         parser = configparser.ConfigParser(delimiters="=")
-        with lando_config_path.open() as f:
-            parser.read_file(f)
+        parser.read_string(lando_ini_contents)
 
-        # If the file doesn't have a `fix` section, exit early.
-        if not parser.has_section("fix"):
-            return None
+        return parser
 
-        fix_hg_command = []
-        for key, value in parser.items("fix"):
-            if not key.endswith(":pattern"):
-                continue
+    def format_stack_tip(self, mach_path: Path) -> Optional[str]:
+        """Add an autoformat commit to the top of the patch stack."""
+        # Update to the tip of the stack.
+        self.run_hg(["update", "-r", "tip"])
 
-            fix_hg_command += ["--config", f"fix.{key}={value}"]
+        # Run linters.
+        subprocess.run([mach_path, "lint", "-r", "stack()"], check=True)
```

Comment:
Wondering if it would be useful to store the command and its output, and put that in the commit message for verbosity?

Reply:
That's not a bad idea - we should definitely at least log the command output.